implement historical cancelBubble attribute for Event interface

yytyd opened the pull request:
<!-- Please describe your changes on the following line: -->
Implementation of `cancelBubble` attribute for Event interface. ref: https://dom.spec.whatwg.org/#dom-event-cancelbubble

---
<!-- Thank you for contributing to Servo! Please replace each `[ ]` by `[X]` when the step is complete, and replace `___` with appropriate data: -->
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #22882 (GitHub issue number if applicable)

<!-- Either: -->
- [x] There are tests for these changes

<!-- Also, please make sure that "Allow edits from maintainers" checkbox is checked, so that we can help you if you get stuck somewhere along the way.-->

<!-- Pull requests that do not address these steps are welcome, but they will require additional verification as part of the review process. -->

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/22928)
<!-- Reviewable:end -->

yytyd committed:
implement historical cancelBubble attribute for Event interface

highfive commented:
Thanks for the pull request, and welcome! The Servo team is excited to review your changes, and you should hear from @nox (or someone else) soon.

highfive commented:
Heads up! This PR modifies the following files:
 * @asajeffrey: components/script/dom/event.rs, components/script/dom/webidls/Event.webidl
 * @kichjang: components/script/dom/event.rs, components/script/dom/webidls/Event.webidl

highfive commented:
<img src="http://www.joshmatthews.net/warning.svg" alt="warning" height=20> **Warning** <img src="http://www.joshmatthews.net/warning.svg" alt="warning" height=20>

* These commits modify script code, but no tests are modified. Please consider adding a test!

KiChjang commented on components/script/dom/event.rs:
```diff
@@ -297,6 +297,18 @@ impl EventMethods for Event {
+        self.stop_propagation.get()
+    }
+
+    // https://dom.spec.whatwg.org/#dom-event-cancelbubble
```
Nit: Link points to the attribute's getter instead of the setter.

KiChjang replied:
Ah, looks like it's impossible to link to the setter instead of the getter. In that case, we should just leave it as `https://dom.spec.whatwg.org/#dom-event-cancelbubble`.

yytyd replied:
Thanks!

KiChjang commented on components/script/dom/event.rs:
```diff
@@ -297,6 +297,18 @@ impl EventMethods for Event {
+    }
+
+    // https://dom.spec.whatwg.org/#dom-event-cancelbubble
+    fn SetCancelBubble(&self, value: bool) -> () {
```
No need for the `-> ()` for a function that returns nothing.

KiChjang reviewed (changes requested):
Thanks for your PR! I have two small nits below. After you made the fixes for them, some WPT expectation ini files will need to be updated as well.

KiChjang commented:
@bors-servo try=wpt

bors-servo commented:
:hourglass: Trying commit b913c6c020ec3135cbf606ffff011a96194a733b with merge 2ff4a9617fed37b095a7310817d8d89863febe36...

KiChjang commented:
There's still an issue reported by Travis:

```
Checking files for tidiness...
./components/script/dom/event.rs:305: method declared in webidl is missing a comment with a specification link
```

KiChjang commented:
@bors-servo try=wpt

bors-servo commented:
:hourglass: Trying commit e7457b9c8bfa621c0c781c8b91f559026ed162dd with merge 9ccde760cc197c507a349d5ea0a428c0585f6d08...

bors-servo commented:
:broken_heart: Test failed - [linux-rel-css](http://build.servo.org/builders/linux-rel-css/builds/11641)

KiChjang commented:
The following tests have unexpectedly passed, and would require updates on the corresponding INI files found in `tests/wpt/metadata`:

```
{
    "status": "PASS", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": "Event interface: attribute cancelBubble", 
    "test": "/dom/interfaces.html?exclude=Node", 
    "line": 88981, 
    "action": "test_result", 
    "expected": "FAIL"
}
{
    "status": "PASS", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": "Event interface: document.createEvent(\"Event\") must inherit property \"cancelBubble\" with the proper type", 
    "test": "/dom/interfaces.html?exclude=Node", 
    "line": 89004, 
    "action": "test_result", 
    "expected": "FAIL"
}
{
    "status": "PASS", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": "Event interface: new Event(\"foo\") must inherit property \"cancelBubble\" with the proper type", 
    "test": "/dom/interfaces.html?exclude=Node", 
    "line": 89029, 
    "action": "test_result", 
    "expected": "FAIL"
}
{
    "status": "PASS", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": "Event interface: new CustomEvent(\"foo\") must inherit property \"cancelBubble\" with the proper type", 
    "test": "/dom/interfaces.html?exclude=Node", 
    "line": 89065, 
    "action": "test_result", 
    "expected": "FAIL"
}
```

ksqsf commented on components/script/dom/event.rs:
```diff
@@ -297,6 +297,17 @@ impl EventMethods for Event {
+    fn SetCancelBubble(&self, value: bool) {
+        if value {
+            self.stop_propagation.set(true)
+        }
```
Why not just `self.stop_propagation.set(value)`?

bors-servo commented:
:umbrella: The latest upstream changes (presumably #22933) made this pull request unmergeable. Please resolve the merge conflicts.

yytyd commented:
@KiChjang Thank you for your support!🙂

I fixed followings:
- Resolved conflicts.
- Added comment to `SetCancelBubble` (for Travis's error).
- Updated `ini` files.

KiChjang commented:
@bors-servo r+

Thanks!

bors-servo commented:
:pushpin: Commit c1e96b4 has been approved by `KiChjang`

<!-- @bors-servo r=KiChjang c1e96b452de522d348a7ae1d55cba746df520fb2 -->

bors-servo commented:
:hourglass: Testing commit c1e96b452de522d348a7ae1d55cba746df520fb2 with merge 0418b32d9514a59fdd8e89278e24b849b8d14831...

bors-servo commented:
:broken_heart: Test failed - [status-taskcluster](https://tools.taskcluster.net/task-group-inspector/#/USFf5iV7RUaeg6Xd7NX0Zw)

KiChjang commented:
@bors-servo retry

```
  ▶ FAIL [expected PASS] /css/CSS2/positioning/abspos-016.xht
  └   → /css/CSS2/positioning/abspos-016.xht 57a99f77207dc80e79319ea5e38564ac91827ec8
```

bors-servo commented:
:bomb: Failed to start rebuilding: `Unknown error`

bors-servo commented:
:hourglass: Testing commit c1e96b452de522d348a7ae1d55cba746df520fb2 with merge fe1663a9f217f0213b7e42ed9bbad6c54b8a47bb...

bors-servo commented:
:sunny: Test successful - [android-mac](http://build.servo.org/builders/android-mac/builds/1154), [arm32](http://build.servo.org/builders/arm32/builds/14030), [arm64](http://build.servo.org/builders/arm64/builds/14058), [linux-rel-css](http://build.servo.org/builders/linux-rel-css/builds/11693), [linux-rel-wpt](http://build.servo.org/builders/linux-rel-wpt/builds/11748), [mac-rel-css1](http://build.servo.org/builders/mac-rel-css1/builds/6635), [mac-rel-css2](http://build.servo.org/builders/mac-rel-css2/builds/6604), [mac-rel-wpt1](http://build.servo.org/builders/mac-rel-wpt1/builds/10846), [mac-rel-wpt2](http://build.servo.org/builders/mac-rel-wpt2/builds/10909), [mac-rel-wpt3](http://build.servo.org/builders/mac-rel-wpt3/builds/6659), [mac-rel-wpt4](http://build.servo.org/builders/mac-rel-wpt4/builds/6592), [magicleap](http://build.servo.org/builders/magicleap/builds/773), [status-taskcluster](https://tools.taskcluster.net/task-group-inspector/#/C7bQsYA5TGWpCqMmc_1lWg)
Approved by: KiChjang
Pushing fe1663a9f217f0213b7e42ed9bbad6c54b8a47bb to master...
